<launch>
  <arg name="COM1" value="/dev/ttyS0"/> 
  <arg name="COM2" value="/dev/ttyS1"/> 
  <arg name="COM3" value="/dev/ttyS2"/> 
  <arg name="COM4" value="/dev/ttyS3"/> 
  <arg name="COM5" value="/dev/ttyS4"/> 
  <arg name="COM6" value="/dev/ttyS5"/> 
  <arg name="COM7" value="/dev/ttyXR0"/> 
  <arg name="COM8" value="/dev/ttyXR1"/> 
  <arg name="COM9" value="/dev/ttyXR2"/> 
  <arg name="COM10" value="/dev/ttyXR3"/> 
  <arg name="COM11" value="/dev/ttyUSB0"/> 
  <arg name="COM12" value="/dev/ttyUSB1"/>
  <arg name="COM13" value="/dev/ttyUSB2"/>
  
  <arg name="BANDRATE_115200" value="115200"/> 
  <arg name="BANDRATE_9600" value="9600"/> 
  <arg name="BANDRATE_2400" value="2400"/> 
  <arg name="BANDRATE_38400" value="38400"/> 
  
  <arg name="DEV_DIR_ADD_LF" value="100"/> 
  <arg name="DEV_DIR_ADD_LB" value="102"/> 
  <arg name="DEV_DIR_ADD_RF" value="101"/> 
  <arg name="DEV_DIR_ADD_RB" value="103"/> 
  <arg name="EXTERN_ADD" value="01"/> 
  <arg name="BMS_ADD" value="01"/> 
  <arg name="DEFAULT_ADD" value="01"/>
  
     <!-- run qtcontroller -->
	<!-- COM10 -->
	<node pkg="laser_ctrol" name="LaserCtrlNode"  type="LaserCtrlNode" output="screen" args="$(arg COM10) $(arg BANDRATE_115200)"/>
	

	 <node pkg="management" name="dgv_manageNode"  type="dgv_manageNode" output="screen"/>

	
    <!-- <node pkg="zmq_ros" name="ZmqProxy"  type="proxy_center" output="screen"/>-->

    <node pkg="sythondriver" name="sythondriver_LF"  type="sythondriver_single" output="screen" args="$(arg DEV_DIR_ADD_LF) $(arg COM3) $(arg BANDRATE_38400) 1" />
    <node pkg="sythondriver" name="sythondriver_LB"  type="sythondriver_single" output="screen" args="$(arg DEV_DIR_ADD_LB) $(arg COM5) $(arg BANDRATE_38400) 1" />
	<node pkg="sythondriver" name="sythondriver_RF"  type="sythondriver_single" output="screen" args="$(arg DEV_DIR_ADD_RF) $(arg COM4) $(arg BANDRATE_38400) -1" />
	<node pkg="sythondriver" name="sythondriver_RB"  type="sythondriver_single" output="screen" args="$(arg DEV_DIR_ADD_RB) $(arg COM6) $(arg BANDRATE_38400) -1" />
	
	<!-- COM11 -->    
	<node pkg="bms" name="BmsNode"  type="BmsNode" output="screen"  args="$(arg COM1) $(arg BANDRATE_9600)" />

	
    <node pkg="joyde" name="JoyTestNode"  type="joyrecvtest" output="screen"/>
	
	<!-- COM2 -->
    <node pkg="extern_sensor" name="ExternNode"  type="ExternNode"  output="screen" args="$(arg EXTERN_ADD) $(arg COM2) $(arg BANDRATE_115200) 1" />
	
    <node pkg="zmq_ros" name="ZmqTransNode"  type="ZmqtransNode" output="screen"/>
	
	<!-- <node pkg="pantiltctrl" name="pantiltctrlNode"  type="pantiltctrlNode" output="screen" /> -->
	
	<!-- COM7 -->
    <!-- <node pkg="remoter" name="RemoterNode"  type="RemoterNode" output="screen" args="$(arg COM7) $(arg BANDRATE_115200)"/> -->
</launch>
